docs/modeling-your-data/modeling-your-data-with-dbt/dbt-operation/backfilling/index.md

```diff
@@ -0,0 +1,114 @@
+
+:::
+
+## Blue-Green style Deployment
+
+As the process of backfilling data may take a long time depending on your data volumes, and during this time no new data will be processed, it may be preferable to backfill your model(s) in a staging environment and then deploy these derived tables into your production schema at a later date. 
+
+:::caution
+
+This process involves manually editing the manifest table, which is dangerous and should be done only with understanding of our [incremental sessionization logic](/docs/modeling-your-data/modeling-your-data-with-dbt/dbt-advanced-usage/dbt-incremental-logic/index.md)
+
+:::
+
+
+Let's assume you have one schema named `prod` that is used by a regularly scheduled run of a project that uses one our package(s), and you want to introduce a new custom model backfilled with all your historic Snowplow data. We are going to first process this model in a `staging` schema before migrating it across.
+
+```mermaid
+%%{init: { 'gantt': {'barHeight': 100, 'fontSize': 30, 'sectionFontSize': 30, 'leftPadding': 200, 'numberSectionStyles':3, 'displayMode': 'compact'}}}%%
+gantt
+    dateFormat  YYYY-MM-DD
+    axisFormat  
+    tickInterval 1day
+    section New Model
+    Unprocessed :crit, n1, 2020-01-01, 5d
+    section Other Models
+    Processed :active, a1, 2020-01-01, 5d
+```
+
+### Step 1: Create and process new model only
+In a local environment, assuming you are using version control for your project, create a new branch and create your model in this branch. Next ensure you are using a profile/target and/or configure the package models so that they will write to the `staging` schema (or be prefixed with `staging` depending on your setup) instead of `prod`. 
+
+Now you will need to run the below script to run only what is needed for your new model (see [specific model selection](/docs/modeling-your-data/modeling-your-data-with-dbt/dbt-operation/model-selection/#specific-model-selection) for more details on why). We've also increased the backfill limit days so this needs to be run fewer times.
+
+
+```bash
+dbt run --select +my_new_model --vars "{'models_to_run': '$(dbt ls --m  +my_new_model --output name)', 'snowplow__backfill_limit_days': 90}"
+```
+
+```mermaid
+%%{init: { 'gantt': {'barHeight': 100, 'fontSize': 30, 'sectionFontSize': 30, 'leftPadding': 200, 'numberSectionStyles':3, 'displayMode': 'compact'}}}%%
+gantt
+    dateFormat  YYYY-MM-DD
+    axisFormat  
+    tickInterval 1day
+    section New Model
+    Unprocessed :crit, n1, after n2, 4d
+    Processed (in staging) :active, n2, 2020-01-01, 1d
+    section Other Models
+    Processed :active, a1, 2020-01-01, 5d
+```
+
+Once you have run this enough times that the data is up-to-date, which you can identify either by looking at the print out in the logs or by querying the manifest table, it's time to move this data to your `prod` schema.
```

Comment:
It could just be a bash script or python that loops the dbt call and scans the logs I guess...

Reply:
Yeah, they could just add a timestamp as a var and it would check the max and equality of all the tables within the manifest and stop when it reaches that time everywhere or sth similar.